src/main/java/slimeknights/tconstruct/tools/common/item/ItemBlockTable.java

```diff
@@ -34,7 +39,40 @@ public void addInformation(@Nonnull ItemStack stack, @Nonnull EntityPlayer playe
     }
 
     if(stack.getTagCompound().hasKey("inventory")) {
-      tooltip.addAll(LocUtils.getTooltips(Util.translate("tooltip.chest.has_items")));
+      // get the inventory, so we can inspect its items
+      NBTTagCompound inventory = stack.getTagCompound().getCompoundTag("inventory");
+
+      if (!inventory.hasKey("Items")) {
+        return;
+      }
+
+      NBTTagList items = inventory.getTagList("Items", 10);
+
+      // check if it's a PatternChest (damage value corresponds to TableTypes enum)
+      if (BlockToolTable.TableTypes.fromMeta(stack.getItemDamage()) == BlockToolTable.TableTypes.PatternChest) {
+        // determine if it holds casts or patterns
+        boolean holdsCasts = false;
+
+        for (int i = 0; i < items.tagCount(); ++i) {
```

Comment:
The chest can only hold either casts or patterns, no need for a loop, the first item is enough.